tests/integration/bond_test.py

```diff
@@ -353,6 +353,71 @@ def test_changing_slave_order_keeps_mac_of_existing_bond(bond99_with_2_slaves):
     )
 
 
+def test_adding_slaves_keeps_mac_of_existing_bond(bond99_with_slave, eth1_up):
+    bond_state = bond99_with_slave[Interface.KEY][0]
+    bond_slaves = bond_state[Bond.CONFIG_SUBTREE][Bond.SLAVES]
+    bond_slaves.insert(0, ETH1)
+
+    current_state = statelib.show_only((BOND99,))
+    assert_mac_address(current_state)
+
+    libnmstate.apply(bond99_with_slave)
+    modified_state = statelib.show_only((BOND99,))
+    assert_mac_address(
+        modified_state, current_state[Interface.KEY][0][Interface.MAC]
+    )
+
+
+def test_adding_to_empty_slaves_keeps_mac_of_existing_bond():
+    with bond_interface(BOND99, []) as state:
+        bond_state = state[Interface.KEY][0]
+        bond_slaves = bond_state[Bond.CONFIG_SUBTREE][Bond.SLAVES]
+
+        current_state = statelib.show_only((BOND99,))
+        assert_mac_address(current_state)
+
+        libnmstate.apply(state)
+        modified_state = statelib.show_only((BOND99,))
+        assert_mac_address(
+            modified_state, current_state[Interface.KEY][0][Interface.MAC]
+        )
+
+
+def test_replacing_slaves_keeps_mac_of_existing_bond(
+    bond99_with_slave, eth1_up
```

Comment:
IMHO it is not very reader-friendly that using `bond99_with_slave` means that `eth2` is the slave. Also it is very dangerous because it is not obvious that the test relies on this.

Reply:
I could do it using `with bond_interface():` but in my opinion we should use existing features. Maybe, we can rename bond99.. but I don't think we should do this in this PR. Would you like another PR?

Reply:
IMHO it is fine to rename it in this PR but probably in a different commit. It could be something like `bond99_with_eth2`.
@EdDev what do you think?

Reply:
If you prefer to do the renaming in a different PR, it is also fine with me.

Reply:
Sounds fine, but you could solve it by depending on both slaves and swapping them without knowing that already exists.
In many tests you do not really care what slave is used, here you do.. this one is an exception.
But I am fine with either ways.

Reply:
Done, thank you!